### PR TITLE
Add checkpoint test for Jakarta Server Faces

### PR DESCRIPTION
**Pull request by dazavala**
For issue #21327

Add a basic Faces (JSF) test case to the checkpoint FAT.  The test case affords more feature coverage as it exercises a complex app, "eventMgr", over two servers that provide a frontend UI (jsf app, mpclient) and backend services (microservice, rest) that manage event data persisted to a derby database.

### CODE REVIEW

**Comment by dazavala** [comment]
#build

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FacesTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                                                   server.waitForStringInLogUsingMark("SRVE0169I: Loading Web Module: frontendUI", 0));
+                                     assertNotNull("'CWWKZ0001I: Application frontendUI started' message not found in log.",
+                                                   server.waitForStringInLogUsingMark("CWWKZ0001I: Application frontendUI started", 0));
+                                 });
+        frontendUI.startServer(getTestMethodNameOnly(testName) + ".log");
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        frontendUI.stopServer();
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FacesTest.java:122]: We need to ensure that both servers stop.  If stopping the `frontendUI` server fails then we will not stop the `backendServices` server.  Perhaps you can have two `@After` methods, one to stop each server?

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FacesTest.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+public class FacesTest {
+
+    public static final String FRONTEND_SERVER_NAME = "checkpointFaces-frontendUI";
+    public static final String BACKEND_SERVER_NAME = "checkpointFaces-backendServices";
+
+    @Rule
+    public TestName testName = new TestName();
+
+//    @ClassRule
+//    public static RepeatTests repeatTest = MicroProfileActions.repeat(FRONTEND_SERVER_NAME, TestMode.LITE, MicroProfileActions.MP41);
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FacesTest.java]: This is commented out for now.  But shouldn't it include MicroProfileActions.MP50 as the other feature sets to run?

```java
    @ClassRule
    public static RepeatTests repeatTest = MicroProfileActions.repeat("timeoutServer", TestMode.LITE, MicroProfileActions.MP41, MicroProfileActions.MP50);

```

File: dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-backendServices/jvm.options [hunk 3]
```diff
@@ -0,0 +1 @@
+-Dcom.ibm.ws.beta.edition=true
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-backendServices/jvm.options]: This doesn't matter, but I'm guessing the backendServices doesn't need beta functionality.

**Review by tjwatson — COMMENTED** [review]
remove the `dev/io.openliberty.checkpoint_fat/test-applications/eventMgr/backendServices/src/java/io/openliberty/guides/event/models/.Event.java.swp` file

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FacesTest.java (hunk 2 above)
Comment (dazavala) [reply dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FacesTest.java]: Yes.  I can certainly add the mp50 feature set to the MPA repeat action.  I removed it while attempting to debug the baseline mp41 repeat action.  FYI on a relate note, we may need to augment MPA to support feature replacement for multiple servers in a repeat action.

File: dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-backendServices/jvm.options (hunk 3 above)
Comment (dazavala) [reply dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-backendServices/jvm.options]: Agree.  May as well remove it now.

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FacesTest.java (hunk 1 above)
Comment (dazavala) [reply dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FacesTest.java:122]: I'll fix this by keeping things consistent: stop `backendServices` in the test that starts it; stop `frontendUI` in the `@After tearDown`

**Comment by dazavala** [comment]
#build
New commit containing review changes rebased on integration.

**Comment by dazavala** [comment]
#libbybot

**Comment by dazavala** [comment]
#build

**Comment by dazavala** [comment]
#libbybot

**Review by tjwatson — APPROVED** [review]

**Pull request merged**